plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -833,8 +833,8 @@ def modify_xml(self):
         self._templatize_pkgid(filelists)
 
         self.repodata['primary'] = utils.remove_fake_element(utils.element_to_text(faked_primary))
-        self.repodata['other'] = utils.remove_fake_element(utils.element_to_text(other))
-        self.repodata['filelists'] = utils.remove_fake_element(utils.element_to_text(filelists))
+        self.repodata['other'] = utils.element_to_text(other)
+        self.repodata['filelists'] = utils.element_to_text(filelists)
```

Comment:
Was `remove_fake_element` just being erroneously (or overzealously, perhaps) applied to `other` and `filelists`? Based on [this](https://pulp.plan.io/issues/2037#note-9) comment, I was expecting something more like replacing a `'str'[:'str',find('foo')]` incantation with `'str'.rstrip()`. Is there a potential landmine still sitting in the code for users of `remove_fake_element`?